### PR TITLE
Conditions for Status reporting

### PR DESCRIPTION
**Pull request by zvonkok**
Added conditions for correct reporting.
Added runAsGroup and a safe group creation

### CODE REVIEW

**Review by ArangoGutierrez — APPROVED** [review]
/lgtm

File: Dockerfile [hunk 1]
```diff
@@ -30,8 +30,8 @@ COPY --from=builder /workspace/manager .
 COPY config/recipes/ /opt/sro/recipes/
 COPY manifests /manifests
 
-RUN useradd -r -u 499 nonroot
-
+RUN useradd  -r -u 499 nonroot
```
Comment (dagrayvid) [inline Dockerfile:33]: Small nit
```suggestion
RUN useradd -r -u 499 nonroot
```

File: Dockerfile [hunk 2]
```diff
@@ -30,8 +30,8 @@ COPY --from=builder /workspace/manager .
...
+RUN getent group nonroot || groupadd -o -g 499 nonroot
```
Comment (dagrayvid) [inline Dockerfile:34]: Two small nits
```suggestion
RUN useradd -r -u 499 nonroot
RUN getent group nonroot || groupadd -o -g 499 nonroot
```

File: controllers/specialresource.go [hunk 3]
```diff
@@ -49,12 +54,23 @@ import (
...
+
 // ReconcilerSpecialResources Takes care of all specialresources in the cluster
 func ReconcilerSpecialResources(r *SpecialResourceReconciler, req ctrl.Request) (ctrl.Result, error) {
 
-	r.Log.Info("Reconciling SpecialResource(s) in all Namespaces")
+	log = r.Log.WithName(prettyPrint("preamble", Purple))
 
+	log.Info("Reconcling ClusterOperator")
+	if err := r.clusterOperatorStatusReconcile(); err != nil {
+		log.Info("Reconcling ClusterOperator failed", "error", fmt.Sprintf("%v", err))
```
Comment (dagrayvid) [inline controllers/specialresource.go:67]: Another nit, for later reference (don't need to hold the merge for this)
```suggestion
	log.Info("Reconciling ClusterOperator")
	if err := r.clusterOperatorStatusReconcile(); err != nil {
		log.Info("Reconciling ClusterOperator failed", "error", fmt.Sprintf("%v", err))
```

File: controllers/status.go [hunk 4]
```diff
@@ -22,3 +29,91 @@ func updateStatus(obj *unstructured.Unstructured, r *SpecialResourceReconciler,
...
+			return nil
+		}
+	}
+	// If we land here there is no clusteroperator object for SRO
+	// Unsure if we need to create a clusteroperator object ...
+	log.Info(prettyPrint("TOOD: Do we need to create a ClusterOperator Object? ", Red))
+	return errs.New("ClusterOperator can not be found")
+}
+
+func (r *SpecialResourceReconciler) clusterOperotarStatusSet() error {
```
Comment (dagrayvid) [inline controllers/status.go:54]:
```suggestion
func (r *SpecialResourceReconciler) clusterOperatorStatusSet() error {
```

File: controllers/status.go [hunk 5]
```diff
@@ -22,3 +29,91 @@ func updateStatus(obj *unstructured.Unstructured, r *SpecialResourceReconciler,
...
+
+func (r *SpecialResourceReconciler) clusterOperatorStatusReconcile() error {
+
+	if err := r.clusterOperatorStatusGetOrCreate(); err != nil {
+		return errs.Wrap(err, "Cannot get or create ClusterOperator")
+	}
+
+	r.clusterOperatorConditionsDefault()
+
+	if err := r.clusterOperotarStatusSet(); err != nil {
```
Comment (dagrayvid) [inline controllers/status.go:102]:
```suggestion
	if err := r.clusterOperatorStatusSet(); err != nil {
```

**Comment by dagrayvid** [comment]
Other than the nits I commented, this looks good to me. Let's just remember to fix these typos post feature-freeze.

**Review by dagrayvid — APPROVED** [review]

**Comment by zvonkok** [comment]
@dagrayvid Thanks, will keep this in mind.

**Comment by dagrayvid** [comment]
Flaky CI test, holding to restart e2e-aws
/hold

**Pull request merged**